.github/workflows/build.yml

```diff
@@ -189,8 +189,11 @@ jobs:
 
           # for supported versions by espressif, see:
           # https://docs.espressif.com/projects/esp-idf/en/latest/esp32/versions.html
-          - v4.0.4
-          - v4.1.2
+          #
+          # to build on the following commented-out relases, port all event loop
+          # handling to one with legacy event loop.
```

Comment:
it turned out that you would need a bit more than that. as the network stack before 4.2 is TCP/IP Adapter, `wireguardif.c` needs more `ifdef`s. i do not think it's worth the effort to support 4.1 because, if you need wireguard in your design, it must be a new design. 4.1 is not for new designs.